src/api/controllers/ReportController.ts

```diff
@@ -0,0 +1,67 @@
+import { Body, CurrentUser, JsonController, Get, Post, Delete, Params} from "routing-controllers";
+import { UserModel } from "../../models/UserModel";
+import { ReportService } from "../../services/ReportService";
+import { ReportPostRequest, ReportProfileRequest, ReportMessageRequest } from "../../types/ApiRequests";
+import { GetReportResponse, GetReportsResponse } from "../../types/ApiResponses";
+import { ReportModel } from "../../models/ReportModel";
+import { report } from "process";
+import { UuidParam } from "../validators/GenericRequests";
+
+@JsonController("report/")
+export class ReportController {
+  private reportService: ReportService;
+
+  constructor(reportService: ReportService) {
+    this.reportService = reportService;
+  }
+
+  @Get("admin/all/")
+  async getAllReports(@CurrentUser() user: UserModel): Promise<GetReportsResponse> {
+    return { reports: await this.reportService.getAllReports(user) };
+  }
+
+  @Get("admin/post/")
+  async getAllPostReports(@CurrentUser() user: UserModel): Promise<GetReportsResponse> {
+    return { reports: await this.reportService.getAllPostReports(user) };
+  }
+
+  @Get("admin/profile/")
+  async getAllProfileReports(@CurrentUser() user: UserModel): Promise<GetReportsResponse> {
+    return { reports: await this.reportService.getAllProfileReports(user) };
+  }
+
+  @Get("admin/message/")
```

Comment:
We should add admin auth middleware here to protect these routes from non admin accounts (i.e., require an access token generated from an admin account auth to be sent along with each of the requests). The middleware can just check to make sure the provided token is valid and comes from an auth account -- we can do this by using the firebase admin sdk on backend: https://firebase.google.com/docs/auth/admin/verify-id-tokens